### PR TITLE
avm1: Add super basic 'ASnative' implementation

### PR DESCRIPTION
**Pull request by Dinnerbone**
It's enough to make `Color-v5` from #22050 pass, at least

### CODE REVIEW

File: tests/tests/swfs/avm1/asnative/output.txt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+[type Function]
+
+// ASnative("onehundred", 0)
+undefined
+
+// ASnative(100, {valueOf: function() {return 4;}})("success")
+success
+
+// ASnative(100, 4.5)("success")
+success
```
Comment (moulins) [inline tests/tests/swfs/avm1/asnative/output.txt:23]: Could you also add tests for `ASnative(100, 0x1_0000 + 4)` and `ASnative(100, 0x1_0000_0000 + 4)`, to check overflow behavior?

Comment (Dinnerbone) [reply tests/tests/swfs/avm1/asnative/output.txt:23]: Done

**Comment by Dinnerbone** [comment]
Color-v5 gnash passes, amended commit to remove its known_failure
